frontend/views/containers/user-settings/AppLogs.vue

```diff
@@ -0,0 +1,178 @@
+      form: {
+        filter: []
+      },
+      ephemeral: {
+        logs: []
+      }
+    }
+  },
+  created () {
+    this.form.filter = this.appLogsFilter
+    sbp('okTurtles.events/on', CAPTURED_LOGS, this.addLog)
+
+    const logs = []
+    let lastEntry = localStorage.getItem('giConsole/lastEntry')
+    do {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${lastEntry}`))
+      if (!entry) break
+      logs.push(entry)
+      lastEntry = entry.prev
+    } while (lastEntry)
+    this.ephemeral.logs = logs.reverse()
+  },
+  beforeDestroy () {
+    sbp('okTurtles.events/off', CAPTURED_LOGS)
+  },
+  watch: {
+    async 'form.filter' (filters) {
+      if (!filters) return
+      this.setAppLogsFilters(filters)
+      sbp('okTurtles.events/emit', SET_APP_LOGS_FILTER, filters)
+    },
+    prettyLogs () {
+      // Automatically scroll textarea to the bottom
+      this.$nextTick(() => {
+        if (this.$refs.textarea) {
+          this.$refs.textarea.scrollTop = this.$refs.textarea.scrollHeight
+        }
+      })
+    }
+  },
+  computed: {
+    ...mapState([
+      'appLogsFilter'
+    ]),
+    prettyLogs () {
+      const isEntryNew = (type) => ['NEW_SESSION', 'NEW_VISIT'].includes(type)
+      return this.ephemeral.logs
+        .filter(({ type }) => {
+          return this.form.filter.includes(type) || isEntryNew(type)
+        })
+        .map(({ type, msg, timestamp }) => {
+          return isEntryNew(type)
+            ? `--------------- \n${timestamp} [${type}]`
+            : `${timestamp} [${type}] ${msg.map(part => JSON.stringify(part)).join(' ')}`
+        })
+        .join('\n')
+    }
+  },
+  methods: {
+    ...mapMutations([
+      'setAppLogsFilters'
+    ]),
+    ...mapActions([
+      'saveSettings'
+    ]),
+    addLog (logHash) {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${logHash}`))
+      if (entry) {
+        this.ephemeral.logs.push(entry)
+      }
+    },
+    openTroubleshooting () {
+      this.$router.push({
+        query: {
+          ...this.$route.query,
+          section: 'troubleshooting'
+        }
+      })
+    },
+    downloadLogs () {
+      downloadLogs('gi_logs.txt', this.$refs.linkDownload)
+    }
+  }
+}
+</script>
+
+<style lang='scss' scoped>
+@import "@assets/style/_variables.scss";
+
+.settings-container {
+  @include desktop {
+    padding-top: 1.5rem;
+  }
+}
+
+.c-header {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.c-filters {
+  flex-grow: 99; // to be wider than c-download.
+
+  .checkbox:last-child {
+    margin-right: 1rem;
+  }
+}
+
+.c-download {
+  flex-grow: 1;
+
+  @include touch {
+    .c-icon {
+      &::before {
+        content: "\f1e0", // .icon-share-alt
+      }
+    }
+  }
+}
+
+.c-filters,
+.c-download,
+.c-logs {
+  margin-bottom: 1rem;
+}
+
+.c-logs {
+  font-family: "Monaco", "Menlo", "Courier", monospace;
+  font-size: $size_5;
+  white-space: pre;
```

Comment:
and the no-wrap thing

Reply:
The "no-wrap thing" is `white-space: pre;` ;) The difference between both is that `pre` respect new lines `\n`, and `no-wrap` doesn't.

Reply:
oh! thanks, didn't realize that!